Formula/texlab.rb

```diff
@@ -0,0 +1,27 @@
+require "language/node"
+
+class Texlab < Formula
+  desc "Implementation of the Language Server Protocol for LaTeX"
+  homepage "https://texlab.netlify.com/"
+  url "https://github.com/latex-lsp/texlab/archive/v1.7.0.tar.gz"
+  sha256 "aa36f3f23656da0d58357fe044d2f60bc01a99bbaa2481132231f4791ff82d17"
+  head "https://github.com/latex-lsp/texlab.git"
+
+  depends_on "node" => :build
+  depends_on "rust" => :build
+
+  def install
+    cd "src/citeproc/js"
+    system "npm", "install", *Language::Node.local_npm_install_args
+    system "npm", "run", "dist"
+    cd "../../.."
+    system "cargo", "install", "--locked",
+                               "--root", prefix,
+                               "--path", "."
+    bin.install "target/release/texlab"
+  end
+
+  test do
+    system bin/"texlab", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Sorry, it said at some point that "build and link correctly" is sufficient and explicitly mentioned `--version` (probably some old text somewhere).

I'll change the test to check the response for a simple LSP initialization request.

Reply:
I've added a test that exercises the function (proper response to a LSP request; I've followed the example from `clojure-lsp.rb`).